default.nix

```diff
@@ -139,8 +139,8 @@ in
   shellHook           = let toYesNo = b: if b then "YES" else "NO"; in ''
     # somehow, CC gets overriden so we set it again here.
     export CC=${stdenv.cc}/bin/cc
-    export GHC=${ghc}/bin/ghc
-    export GHCPKG=${ghc}/bin/ghc-pkg
+    ${lib.optionalString (! withHadrianDeps) "export GHC=${ghc}/bin/ghc"}
```

Comment:
Sorry, but I have to admit that I'm not convinced of the change. (Please feel free to convince me :smile: )

I could think of the following scenarios:
- `GHC` isn't needed at all. In this case we should remove it.
- `GHC` is needed. In this case we should use a GHC that has the needed packages added.

When under some circumstance `GHC` is set and under some not, this seems to be confusing to users... :thinking:

Reply:
I agree that this might not be the best solution for all the way ghc can be enabled/disabled. I'll try to think of something better. It was kind of a quick fix because it bothers me when cabal is fetching deps on it's own :joy:

Reply:
Sorry, now that I fould some time to look into this, I'm confused.

> GHC isn't needed at all. In this case we should remove it.

From where should we remove GHC. I though one of the points of `ghc.nix` is to make GHC available.

> GHC is needed. In this case we should use a GHC that has the needed packages added.

We do use a GHC with the needed packages. But for the Hadrian scenario `./configure` finds ghc in PATH and not from GHC.

Reply:
I think I found a simpler solution I hope we can agree on. ☺

Reply:
@neosimsim thanks a lot for taking the time to consider my complaints. :smile: 

With your change the environment variable `GHC` now points to `ghc` from the nix env's path. I think that's what should be accomplished. :+1: 

I still don't understand why we need the environment variable `GHC` at all. But, I'm too afraid to just remove it (don't want to break `ghc.nix` for any users/use cases).